test/extended/cmd.sh

```diff
@@ -71,75 +72,69 @@ popd
 rm -rf $tmp
 
 # ensure a local-only image gets a docker image(not imagestream) reference created.
-[ "$(oc new-app test/scratchimage~https://github.com/openshift/ruby-hello-world.git --strategy=docker -o yaml |& tr '\n' ' ' | grep -E "from:\s+kind:\s+DockerImage\s+name:\s+test/scratchimage:latest\s+")" ]
+os::cmd::expect_success_and_text "oc new-app test/scratchimage~https://github.com/openshift/ruby-hello-world.git --strategy=docker -o yaml |& tr '\n' ' '" "from:\s+kind:\s+DockerImage\s+name:\s+test/scratchimage:latest\s+"
 # error due to partial match
-[ "$(oc new-app test/scratchimage2 -o yaml |& tr '\n' ' ' 2>&1 | grep -E "partial match")" ]
+os::cmd::expect_failure_and_text "oc new-app test/scratchimage2 -o yaml |& tr '\n' ' '" "partial match"
```

Comment:
not sure the tr is needed anymore here, given we're just checking for a single string that should be on one line.